### ISSUE TITLE
Ignore or select which field to lock ?

### ISSUE THREAD

**Issue by Jerome1337** [doctrine/doctrine2#5610]
I'm using lock protection on my `User` entity, this entity extend FOSUserBundle `User` entity.

But I have some problems when there is a lot of request on the same user (API request etc) and this cause `500 Internal Server Error `

There is any way to ignore some field on this entity ?
Can i choose which field i want to lock or ignore with Doctrine Optimistic Locking ?

Thanks

**Comment by DHager** [comment]
First, exactly what exception is causing your HTTP-500 error?

**Comment by Soullivaneuh** [comment]
> Is it happening because you have two different requests trying to modify the user at the same time?

Exactly. For the concrete case, we are trying to use lock protection with sonata admin on user entity that works perfectly well.

But we have two field (`last_activity` and `last_login`) that are updated on near every request for the one, and on authentication for the second (FOS listener).

We have an API that can be access by user with a specific token. For some reason, the API could be called ~400 times at the same moment with the same user.

So, Doctrine is trying to modify those fields at the same time and throw a `OptimisticLockException`.

I'm not sure it's a bug as @Ocramius tagged on this issue but I think that adding an option to ignore some field for protection would be a nice to have.

**Comment by Ocramius** [comment]
Locking is done at record level. I think the design of the bundle should append to an audit log, insye

**Comment by Soullivaneuh** [comment]
> Locking is done at record level. I think that the bundle should append to an audit log, instead

Not sure to understand well. Could you please elaborate?

**Comment by JCMais** [comment]
I think @Ocramius  means that the updates are locking the user entry in the users table, because you are updating fields of the same record, instead, the bundle should add new records to a "``userActivity``" kind of table.

**Comment by DHager** [comment]
> adding an option to ignore some field for protection would be a nice to have

Hmmm, what should happen when both a "normal" property and a "non-locked" property have been changed and are being saved, and a conflict occurs that throws an `OptimisticLockException`? Should half the change go through anyway, or nothing at all?

**Comment by Ocramius** [comment]
Exactly that, @JCMais, thanks :-)

**Comment by DHager** [comment]
While it may be technically possible to add a "Avoid locking if **all** altered fields are on the ignore-list", my gut feeling is that it isn't worth the added complexity, especially since it might "suddenly stop working" when a version-sensitive field happens to be touched.
